Source/hydro/reconstruction.H

```diff
@@ -78,5 +82,69 @@ load_passive_stencil(Array4<Real const> const& U_arr, Array4<Real const> const&
 
 }
 
-#endif
+AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+void
+add_geometric_rho_source(amrex::Array4<amrex::Real const> const& q_arr,
+                         amrex::Array4<amrex::Real const> const& dloga,
+                         const int i, const int j, const int k,
+                         amrex::Real* s) {
+
+    using namespace reconstruction;
+
+    // this takes the form: -alpha rho u / r
+    // where alpha = 1 for cylindrical and 2 for spherical
+
+    // note: this is assumed to be working only in the x-direction
+
+    s[im2] += -dloga(i-2,j,k) * q_arr(i-2,j,k,QRHO) * q_arr(i-2,j,k,QU);
```

Comment:
are we not multiplying the source by dt/2 ?

Reply:
This is just adding the geometric sources to the other hydrodynamic sources.  Then we reconstruct and integrate under the parabola, and then the dt2 is applied when we add it to the jumps that are accumulated to the interfaces